구매자 마이페이지 UI 구현

1000hyehyang opened the pull request:
## ⭐️ Issue Number
- #17 

## 🚩 Summary
![image](https://github.com/user-attachments/assets/e0003b02-9a4f-43da-b4b0-f9975ba11bc7)
![image](https://github.com/user-attachments/assets/5ea01a0c-f87e-4d84-b621-8eb628a9b415)
![image](https://github.com/user-attachments/assets/778adeb4-cc6a-43a7-ad60-28d8f1ee3074)

- 마이페이지 UI 구현: 프로필 카드, 주문/배송 현황, 사이드바 네비게이션 등 핵심 기능 구현
- 반응형 레이아웃 완성: 데스크탑(2단), 태블릿/모바일(1단) 대응
- 접근성 개선: ARIA 라벨, 키보드 네비게이션, 스크린 리더 호환성 향상
- Mock 데이터 중앙 관리: src/data/mypage.ts에서 데이터 일원화 관리

## 🛠️ Technical Concerns

- 반응형 여백: 모바일 24px, 태블릿 36px, 데스크탑 48px
- 아이콘 매핑 시스템: 동적 아이콘 렌더링으로 유지보수성 향상
- 조건부 렌더링: lg:block, lg:hidden으로 디바이스별 최적화
- Tailwind CSS: 일관된 유틸리티 클래스 사용
- 컴포넌트 분리로 번들 크기 최적화
- 불필요한 리렌더링 방지를 위한 적절한 구조 설계

## 🙂 To Reviewer
@taehyun32 
가볍게 쓱 한 번 봐주세요~

## 📋 To Do

- [ ] 구매자 프로필 수정 UI 구현

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 사용자 프로필, 주문/배송 현황, 쇼핑 활동, 사이드바 등으로 구성된 마이페이지가 추가되었습니다.
  * 데스크톱과 모바일 환경에 맞는 반응형 레이아웃과 모바일 전용 사이드바가 제공됩니다.
  * 아바타, 뱃지, 배지, 카드, 구분선 등 다양한 UI 컴포넌트가 도입되었습니다.
  * 마이페이지용 목업 데이터가 적용되어 프로필 정보와 메뉴가 표시됩니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

1000hyehyang committed:
feat: 마이페이지 UI 구현

- 프로필 카드 (아바타, 닉네임, 뱃지, 포인트)
- 주문/배송 현황 카드
- 사이드바 네비게이션 (데스크탑/모바일)
- 반응형 레이아웃 완벽 대응

- ARIA 라벨 및 role 속성 추가
- 키보드 네비게이션 지원
- 스크린 리더 호환성

1000hyehyang committed:
feat: 마이페이지 메타데이터 추가

taehyun32 reviewed (approved):
와이어프레임에서 봤던 구조가 자연스럽게 잘 녹아든 것 같습니다.
전체적으로 흐름도 이해하기 편하고, 레이아웃도 깔끔하게 잘 잡혀 있는거 같습니다. 수고하셨습니다 :)

1000hyehyang committed:
fix: CardTitle 컴포넌트의 타입 불일치 수정

- forwardRef 제네릭 타입을 HTMLParagraphElement에서 HTMLHeadingElement로 변경
- ref 타입과 실제 DOM 요소(<h3>) 타입 일치시킴
- 타입 안전성 개선

1000hyehyang commented on src/components/ui/card.tsx:
```diff
@@ -0,0 +1,56 @@
+      {...props}
+    />
+  ),
+);
```
OK 반영했음 토끼야

1000hyehyang committed:
fix: Separator 컴포넌트의 WCAG 접근성 개선

- decorative=true일 때 role 제거하여 스크린리더에서 무시되도록 수정
- decorative=false일 때만 role="separator" 설정
- WCAG 가이드라인 준수

1000hyehyang commented on src/components/ui/separator.tsx:
```diff
@@ -0,0 +1,27 @@
+    <div
+      role={decorative ? 'separator' : undefined}
+      aria-orientation={orientation}
+      className={cn(
```
그래 수정했다

1000hyehyang committed:
fix: OrderStatus 컴포넌트 데이터 접근 경로 수정

- orderStats → orderStatuses로 데이터 접근 경로 변경
- 하드코딩된 statusItems 배열 제거하고 실제 데이터 구조 활용
- 데이터 구조와 일치시켜 런타임 오류 방지

1000hyehyang commented on src/components/mypage/OrderStatus.tsx:
```diff
@@ -0,0 +1,35 @@
+    { label: '배송중', count: orderStats.shipping },
+    { label: '배송 완료', count: orderStats.completed },
+    { label: '취소/반품', count: orderStats.cancelled },
+  ];
```
고맙다 수정했다

1000hyehyang committed:
refactor: ProfileCard 컴포넌트 데이터 소스 중앙화

- 하드코딩된 더미 데이터 제거
- myPageData.profile 사용으로 데이터 일관성 확보
- 유지보수성 및 타입 안전성 개선

1000hyehyang commented on src/components/mypage/ProfileCard.tsx:
```diff
@@ -0,0 +1,62 @@
+  avatar: '/ururu-full-logo.png', // public 폴더 내 기본 이미지 사용
+  badges: ['민감성', '여름쿨톤'],
+  point: 12345,
+};
```
고맙다.. 왜 리팩토링 해놓고 안 지웠을까.. 감사요

1000hyehyang committed:
refactor: ShoppingActivities 컴포넌트에서 사용하지 않는 MapPin import 제거

- MapPin 아이콘이 실제로 사용되지 않아 불필요한 import 제거
- 코드 정리 및 번들 크기 최적화

1000hyehyang commented on src/components/mypage/ShoppingActivities.tsx:
```diff
@@ -0,0 +1,78 @@
+      { icon: 'FileText', label: '서비스 이용약관', href: '/terms' },
+    ],
+  },
+];
```
반영했다

1000hyehyang committed:
remove: 사용하지 않는 마이페이지 레이아웃 제거

1000hyehyang committed:
fix: Separator 컴포넌트 접근성 개선

- aria-orientation을 조건부로 설정

1000hyehyang commented on src/components/ui/separator.tsx:
```diff
@@ -0,0 +1,27 @@
+      role={decorative ? 'separator' : undefined}
+      aria-orientation={orientation}
+      className={cn(
+        'bg-border shrink-0',
+        orientation === 'horizontal' ? 'h-px w-full' : 'h-full w-px',
+        className,
+      )}
```
반영했다

1000hyehyang commented on src/components/mypage/ShoppingActivities.tsx:
```diff
@@ -0,0 +1,61 @@
+                    <span className="flex-1 text-sm font-medium text-gray-800">{item.label}</span>
+                    <ChevronRight className="h-4 w-4 text-gray-300 group-hover:text-pink-400" />
+                  </a>
+                </li>
```
나중에 할게

1000hyehyang committed:
refactor: ShoppingActivities 타입 안전성 및 성능 개선

- 구체적인 타입 정의 추가
- PointIcon 컴포넌트 분리

1000hyehyang committed:
style: 사이드바 폰트 굵기 semibold로 변경
